atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/client/creation/DefaultCqlClientFactory.java

```diff
@@ -77,18 +79,26 @@ public CqlClient visit(CassandraServersConfigs.CqlCapableConfig cqlCapableConfig
                     return ThrowingCqlClientImpl.SINGLETON;
                 }
 
-                log.info("Constructing a full CQL client");
-
                 Set<InetSocketAddress> servers = cqlCapableConfig.cqlHosts();
 
-                Cluster cluster = buildCluster(
-                        config,
-                        servers,
-                        loadBalancingPolicy(config, servers),
-                        poolingOptions(config),
-                        queryOptions(config),
-                        cqlCapableConfig.socksProxy().map(SocksProxyNettyOptions::new),
-                        sslOptions(config));
+                Cluster.Builder clusterBuilder = cqlClusterBuilderFactory.get()
+                        .addContactPointsWithPorts(servers)
+                        .withCredentials(config.credentials().username(), config.credentials().password())
+                        .withCompression(ProtocolOptions.Compression.LZ4)
+                        .withRetryPolicy(DefaultRetryPolicy.INSTANCE)
+                        .withoutJMXReporting()
+                        .withProtocolVersion(ProtocolVersion.V3)
+                        .withThreadingOptions(new ThreadingOptions());
+
+                Cluster cluster =
+                        withLoadBalancingPolicy(
+                                withQueryOptions(
+                                        withPoolingOptions(
+                                                withSslOptions(clusterBuilder, config),
+                                                config),
+                                        config),
+                                config,
+                                servers).build();
```

Comment:
This is correct. Though I'd suggest re-assigning the builders to the variable to avoid this much nesting

Reply:
Refactored.